src/test/java/io/thinkit/edc/client/connector/EdcConnectorClientTest.java

```diff
@@ -24,4 +24,26 @@ void should_thrown_exception_when_requesting_assets_but_no_management_url_specif
 
         assertThatThrownBy(client::assets).isInstanceOf(IllegalArgumentException.class);
     }
+
+    @Test
+    void should_register_extension_resources() {
+        var client = EdcConnectorClient.newBuilder()
+                .extendWith(ExtensionService.class, ExtensionService::new)
+                .build();
+
+        var extensionService = client.service(ExtensionService.class);
+
+        assertThat(extensionService.doStuff()).isEqualTo("done");
+    }
```

Comment:
I don't fully understand how `ExtensionService::new` returns the `ResourceCreator` interface - is this something intrinsic to Java? How does it works?

Thank you in advance

Reply:
`<className>::new` is a method reference, is a contracted shape of a lambda, in this case it is equivalent to `context -> new ExtensionService(context)`, that, in fact, is the same signature of the `ResourceCreator` interface method (accepts an `EdcClientContext`, returns a `EdcResource`)

Reply:
In java a constructor is still a method, so

```java
public interface ResourceCreator {
    EdcResource create(EdcClientContext context);
}
```

has the same boundary as:

```java
public class EdcResource {
    public EdcResource(EdcClientContext context) {
    
    }
}
```

so the second can be considered an implementation of the first.
at first can look weird, but in fact it's quite useful.

Reply:
Nice! Lovely, that makes total sense now.

Thank you, @ndr-brt